here-naksha-lib-handlers/src/main/java/com/here/naksha/lib/handlers/internal/AdminFeatureEventHandler.java

```diff
@@ -99,17 +93,17 @@ abstract class AdminFeatureEventHandler<FEATURE extends NakshaFeature> extends A
   /**
    * Direct validation of XyzFeature to be written. It's optional to implement this, by default it always succeeds.
    *
-   * @param feature feature to be validated before being written
+   * @param codec containing the feature to be validated before being written
    * @return validation result, success by default
    */
-  protected @NotNull Result validateFeature(FEATURE feature) {
+  protected @NotNull Result validateFeature(XyzFeatureCodec codec) {
```

Comment:
Why not `validateFeature(FEATURE feature, EWriteOp op)`?
If we will rely on `XyzFeatureCodec` then subclasses (like `IntStorageHandler`) won't have their contract limited to just their type (for exmaple: `IntStorageHandler` will accept **any** `XyzFeatureCode` while it could just accept `Storage` along with op).

Reply:
These classes are one of few places where we could actually get benefit of statically typed language - let's use types as narrowly as possible :)

Reply:
Discussed with @gunplar offline - we're leaving this for now as this was discussed between @gunplar and @hirenkp2000 .
That being said, I might propose similar alternative soon in my PR - not to block this one

Reply:
Well, since we need access to feature Id, WriteOp and an entire Feature object inside `validateFeature` , I can understand `FeatureCodec` to be a logical choice.
However, my suggestion to @gunplar was to evaluate this further with you (@Amaneusz), as the same problem will be applicable for Delete handler request as well and it wouldn't be good to have 2 different approaches between Storage and Handler. Reading the comments above, it looks like that discussion hasn't been conclusive.
Nevertheless, I agree with @Amaneusz that this is not a critical point to block PR. If Jakub thinks of better alternative approach, it can be addressed in the next PR as well.